app/src/main/java/org/oppia/app/hintsandsolution/HintsViewModel.kt

```diff
@@ -37,33 +37,36 @@ class HintsViewModel @Inject constructor() : HintsAndSolutionItemViewModel() {
     for (index in hintList.indices) {
       if (itemList.isEmpty()) {
         addHintToList(hintList[index])
-      } else {
+      } else if (itemList.size > 1) {
         val isLastHintRevealed =
-          (itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get()!!
+          (itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get() ?: false
         if (isLastHintRevealed && index <= newAvailableHintIndex.get()!! / 2) {
           addHintToList(hintList[index])
         } else {
           break
         }
       }
     }
-    val isLastHintRevealed = (itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get()!!
-    if (solution.hasExplanation() &&
-      hintList.size * 2 == itemList.size &&
-      isLastHintRevealed &&
-      allHintsExhausted.get()!!
-    ) {
-      addSolutionToList(solution)
+    if (itemList.size > 1) {
+      val isLastHintRevealed =
+        (itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get() ?: false
+      if (solution.hasExplanation() &&
+        hintList.size * 2 == itemList.size &&
+        isLastHintRevealed &&
+        allHintsExhausted.get()!!
```

Comment:
Ditto for these observables (prefer not using !! operator). I suggest doing a pass on this file to remove all such cases (local vals can help reduce the logical complexity increase that can come from adding these checks in a bunch of places).

Reply:
Solved.